src/asound/bluealsa-ctl.c

```diff
@@ -601,6 +783,9 @@ static void bluealsa_close(snd_ctl_ext_t *ext) {
 		free(ctl->dev_list[i]);
 	free(ctl->dev_list);
 	free(ctl->pcm_list);
+	for (i = 0; i < ctl->elem_list_size; i++)
+		if (ctl->elem_list[i].type == CTL_ELEM_TYPE_CODEC)
```

Comment:
Calling `free()` on `NULL` is a no-op, so this if is redundant here.

Reply:
The codecs field is not initialized to NULL when unused. I could add that initialization and then remove this if() statement if you prefer.